server/channels/api4/channel.go

```diff
@@ -169,6 +169,10 @@ func updateChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 			c.Err = model.NewAppError("updateChannel", "api.channel.patch_update_channel.forbidden.app_error", nil, "", http.StatusForbidden)
 			return
 		}
+		if (channel.Name != "" && channel.Name != oldChannel.Name) || (channel.DisplayName != "" && channel.DisplayName != oldChannel.DisplayName) || (channel.Purpose != oldChannel.Purpose) {
```

Comment:
Why can't we change the purpose? @larkox

Reply:
That would be a question for product. :P

As far as I know, there is no way on the UI to change or see the purpose of a DM/GM, so... it shouldn't be allowed to change.

Reply:
I think it should be fine. The server in general should be more accepting of custom clients doing their own thing. If there is no reason for a limitation, it should be fine to allow clients to change fields that the UI doesn't support. 

@johndavidlugtu - do we want to intentionally block clients from updating the `purpose` field for DM/GMs?

Reply:
@agnivade I'm leaning towards keeping things consistent and not allowing ability to update the purpose field of a DM/GM.

Rationale being, as we move towards making clearer the distinction between a DM/GM vs channels, we want to reduce overlap where possible. DMs and GMs are supposed to be ephemeral and/or used less frequently compared to channels.

As we currently don't allow in the UI to change the purpose of DM/GM, let's keep it as is. That said, if we hear feedback from users, we can reconsider in the future.

Still open to thoughts if you have anything more to add

Reply:
Got it. Sounds good to me.